### ISSUE TITLE
Allies become puppets 

### ISSUE THREAD

**Issue by Dhaonu** [The-Grand-Combination/The-Grand-Combo#348]
So I am spain, waged ear to get Gibraltar and got almost 82 war score. I selected place in the sun, reparation,  a couple of others and also release and puppet bengal. After they accept the peace, ALL MY ALLIES, all of them, become mt puppets (with the exception of germany ofc). Is there something to it?

**Comment by Dhaonu** [comment]
Dude, why mark it as completed? That was a legitimate issue

**Comment by pixelaes** [comment]
It got fixed; the 'Release and Puppet' CB had a flawed check that would include all spherelings, thereby turning all spherelings into your puppets. That bug got fixed when the bugfixes branch got merged into master.

**Comment by Dhaonu** [comment]
Thank you man. So now I just need to updat tgc right?

**Comment by pixelaes** [comment]
Yup, updating from master and loading a save from before you enforced the peace deal should fix it.

**Comment by Dhaonu** [comment]
Sorry to be noob. That means I need to download the great combo again and copy replace in the mod folder. Or is it a smaller update?

Thank you

**Comment by pixelaes** [comment]
Downloading the whole mod again is an option, but if you go to mod/TGC/common, you can replace the cb_types.txt file that you currently have with this one, which should fix the issue.

[cb_types.txt](https://github.com/user-attachments/files/19438953/cb_types.txt)

**Comment by Dhaonu** [comment]
Top man.

Thank you very much
